File: ChoreAssignmentTracker/src/Chores.Core/Models/Chore.cs
namespace Chores.Core.Models;

public class Chore
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int PointValue { get; set; }
    public string? Frequency { get; set; }
    public int? EstimatedMinutes { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    public Guid? TenantId { get; set; }
}

File: ChoreAssignmentTracker/src/ChoreAssignmentTracker.WebApp/projects/chore-assignment-tracker-shared/src/lib/models/chore.ts
import { ChoreFrequency } from './chore-frequency';

export interface Chore {
  choreId: string;
  userId: string;
  name: string;
  description?: string;
  frequency: ChoreFrequency;
  estimatedMinutes?: number;
  points: number;
  category?: string;
  isActive: boolean;
  createdAt: Date;
}

export interface CreateChore {
  userId: string;
  name: string;
  description?: string;
  frequency: ChoreFrequency;
  estimatedMinutes?: number;
  points: number;
  category?: string;
}

export interface UpdateChore {
  name: string;
  description?: string;
  frequency: ChoreFrequency;
  estimatedMinutes?: number;
  points: number;
  category?: string;
  isActive: boolean;
}
